packages/xrpl/test/integration/transactions/permissionedDEX.test.ts

```diff
@@ -0,0 +1,373 @@
+    'Validate the properties of the DirectoryNode ledger object (contains the PermissionedDEX Offer object)',
+    async () => {
+      const directoryNode_ledger_object = await testContext.client.request({
+        command: 'ledger_entry',
+        directory: offer_ledger_object.BookDirectory,
+      })
+
+      assert.equal(
+        (directoryNode_ledger_object.result.node as DirectoryNode).index,
+        offer_ledger_object.BookDirectory,
+      )
+      assert.equal(
+        (directoryNode_ledger_object.result.node as DirectoryNode)
+          .LedgerEntryType,
+        'DirectoryNode',
+      )
+      assert.equal(
+        (directoryNode_ledger_object.result.node as DirectoryNode).DomainID,
+        pd_ledger_object.index,
+      )
+    },
+    TIMEOUT,
+  )
+
+  it(`Validate the bookOffers method`, async () => {
+    const response: BookOffersResponse = await testContext.client.request({
+      command: 'book_offers',
+      taker: wallet2.classicAddress,
+      taker_pays: {
+        currency: 'USD',
+        issuer: testContext.wallet.classicAddress,
+      },
+      taker_gets: {
+        currency: 'XRP',
+      },
+      domain: pd_ledger_object.index,
+    } as BookOffersRequest)
+
+    assert.equal(response.result.offers.length, 1)
+    assert.equal(response.result.offers[0].TakerGets, '1000')
+    assert.equal(
+      (response.result.offers[0].TakerPays as IssuedCurrencyAmount).value,
+      '10',
+    )
+    assert.equal(
+      (response.result.offers[0].TakerPays as IssuedCurrencyAmount).currency,
+      'USD',
+    )
+    assert.equal(
+      (response.result.offers[0].TakerPays as IssuedCurrencyAmount).issuer,
+      testContext.wallet.classicAddress,
+    )
+
+    assert.equal(response.result.offers[0].DomainID, pd_ledger_object.index)
+  })
+
+  it(`Validate the subscription stream for PermissionedDEX offers`, async () => {
+    const subscribe_request: SubscribeRequest = {
+      command: 'subscribe',
+      books: [
+        {
+          taker_gets: { currency: 'XRP' },
+          taker_pays: {
+            currency: 'USD',
+            issuer: testContext.wallet.classicAddress,
+          },
+          taker: wallet1.classicAddress,
+          domain: pd_ledger_object.index,
+        } as SubscribeBook,
+      ],
+    }
+
+    const response = await testContext.client.request(subscribe_request)
+    assert.equal(response.type, 'response')
+
+    // Note: The result is empty because no Offer has been created after the creation of the Subscription stream.
+    // This case is tested in the rippled code. To avoid the additional complexity, validating the contents
+    // of the response is skipped in this test.
+    // This test validates that domain_id is an acceptable input parameter to the subscribe command.
+    assert.isEmpty(response.result)
+  })
+
+  it(
+    'Crossing a PermissionedDEX Offer',
+    async () => {
+      // wallet2 "crosses" the offer within the domain
+      const offerCrossTx: OfferCreate = {
```

Comment:
```suggestion
      const offerCreateTx: OfferCreate = {
```

Reply:
It's confusing because there is no transaction named `OfferCross`. The comment above it describes it's use too.

Reply:
75ae1c3d7